src/test/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImplTest.java

```diff
@@ -0,0 +1,256 @@
+        externalObjectDirectoryEntity.setId(1);
+        externalObjectDirectoryEntity.setStatus(EodHelper.armProcessingResponseFilesStatus());
+        externalObjectDirectoryEntity.setTransferAttempts(1);
+        objectStateRecordEntity = new ObjectStateRecordEntity();
+        objectStateRecordEntity.setUuid(123L);
+        externalObjectDirectoryEntity.setOsrUuid(objectStateRecordEntity.getUuid());
+
+    }
+
+    @Test
+    void getManifestFilePrefix() {
+        // given
+        when(configuration.getDetsManifestFilePrefix()).thenReturn("DETS");
+
+        // when
+        String result = detsToArmBatchProcessResponseFilesImpl.getManifestFilePrefix();
+
+        // then
+        assertNotNull(result);
+        verify(configuration, times(1)).getDetsManifestFilePrefix();
+    }
+
+    @Test
+    void preProcessResponseFilesActions() {
+        // given
+        when(osrRepository.findByArmEodId(String.valueOf(externalObjectDirectoryEntity.getId()))).thenReturn(Optional.of(objectStateRecordEntity));
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl.preProcessResponseFilesActions(externalObjectDirectoryEntity.getId());
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(String.valueOf(externalObjectDirectoryEntity.getId()));
+        verify(osrRepository, times(1)).save(objectStateRecordEntity);
+    }
+
+    @Test
+    void onUploadFileChecksumValidationSuccess() {
+        // given
+        ArmResponseUploadFileRecord uploadFileRecord = new ArmResponseUploadFileRecord();
+        uploadFileRecord.setA360FileId("a360FileId");
+        String checksum = "checksum";
+        uploadFileRecord.setMd5(checksum);
+        ArmResponseCreateRecord armResponseCreateRecord = new ArmResponseCreateRecord();
+
+        ArmResponseBatchData batchData = ArmResponseBatchData.builder()
+            .externalObjectDirectoryId(externalObjectDirectoryEntity.getId())
+            .armResponseCreateRecord(armResponseCreateRecord)
+            .createRecordFilenameProcessor(createRecordFilenameProcessor)
+            .armResponseUploadFileRecord(uploadFileRecord)
+            .uploadFileFilenameProcessor(uploadFileFilenameProcessor)
+            .build();
+        when(osrRepository.findByArmEodId(anyString())).thenReturn(Optional.of(objectStateRecordEntity));
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl.onUploadFileChecksumValidationSuccess(batchInputUploadFileFilenameProcessor,
+                                                                                     batchData,
+                                                                                     uploadFileRecord,
+                                                                                     externalObjectDirectoryEntity,
+                                                                                     checksum);
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(anyString());
```

Comment:
Verify should use expected values instead of any just to make sure it was hit with the right things

Reply:
done